### PR TITLE
Hook up extract local var refactoring

### PR DESCRIPTION
**Pull request by danrubel**
@devoncarew 
- generalize check / perform / apply refactoring methods
- hook up extract local variable refactoring

### CODE REVIEW

File: menus/atom-dart.cson [hunk 1]
```diff
@@ -36,6 +36,8 @@
     {label: 'Format', command: 'dartlang:dart-format' }
     {label: 'Organize Directives', command: 'dartlang:organize-directives' }
     {label: 'Rename…', command: 'dartlang:refactor-rename' }
+    # TODO(danrubel) should this be placed in a Refactor submenu ?
```
Comment (devoncarew) [inline menus/atom-dart.cson]: +1, or break the menu items up into those that modify code, and ones that are more information. I'll open an issue to track this.

Comment (devoncarew) [inline menus/atom-dart.cson]: https://github.com/dart-atom/dartlang/issues/595

File: lib/analysis/refactor.dart [hunk 2]
```diff
@@ -15,153 +15,212 @@ import '../utils.dart';
...
+  void _handleExtractLocal(String path, int offset, int end, String text) {
+    // Check if extract local can be performed
+    _checkRefactoringAvailable(
+        Refactorings.EXTRACT_LOCAL_VARIABLE, path, offset, end,
+        (AvailableRefactoringsResult result) {
+      // TODO: use the rename refactoring feedback
+      // to better select the ID being renamed.
+      String oldName = '';
+
+      promptUser('Extract local variable: enter the new name.',
```
Comment (devoncarew) [inline lib/analysis/refactor.dart]: Perhaps, `Extract local variable - enter variable name:`?

File: lib/analysis/refactor.dart [hunk 3]
```diff
@@ -15,153 +15,212 @@ import '../utils.dart';
...
 
-    // TODO: Timeout if the refactor request takes too long?
-    Job job = new AnalysisRequestJob('rename', () {
+    // Check if rename can be performed
+    _checkRefactoringAvailable(Refactorings.RENAME, path, offset, end,
+        (AvailableRefactoringsResult result) {
+      // TODO: use the rename refactoring feedback
+      // to better select the ID being renamed.
+
+      promptUser('Rename refactor: enter the new name.',
```
Comment (devoncarew) [inline lib/analysis/refactor.dart]: Similarly, `Rename refactor - enter the new name:`?

**Comment by devoncarew** [comment]
lgtm! Two suggestions for verbiage.

File: lib/analysis/refactor.dart (hunk 2 above)
Comment (danrubel) [inline lib/analysis/refactor.dart]: Sgtm. Done.

**Pull request merged**